lib/Pingmachine/Probe/HTTPing.pm

```diff
@@ -194,8 +203,21 @@ sub _collect_current_job {
             if ($line =~ /could not connect|time out|short read/) {
                 push @pings, undef;
             }
+            # if http codes to be considered as failures are defined and the line contains one of them, ignore the rtt value and add a -
+            elsif ($self->http_codes_as_failure) {
+                # replace for example 403,407 with 403 [A-Z][a-z]+| 407 [A-Z][a-z]+ to match response codes 403 Forbidden, 407 Proxy Authentication Required
+                my $httping_errors .= $self->http_codes_as_failure =~ s/,/ [A-Z]+[a-z]*|/gr;
+                $httping_errors .=  ' [A-Z]+[a-z]*';
```

Comment:
Are you sure this will match what it should?
Say we have `403,407`, the two lines above will turn this into `403 [A-Z]+[a-z]*|407 [A-Z]+[a-z]*`.
This would match for example `403 Forbidden` but it would not match `407 Proxy Authentication Required` due to the whitespaces, no? It does match `407 Proxy` though. Is that enough?

Reply:
Maybe this could be turned into something like `(403|407) [A-Z]+[a-z]*(\s[A-Z]+[a-z]*)?`?
i.e. split the codes on the `,` into the `(code1|code2|code3)` and then add one generic string match at the end?

Reply:
my idea, based on httping output, was that a match with 407 Proxy was enough but your comment makes perfectly sense. PR has been updated.